alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/ConnectionSocket.java

```diff
@@ -86,22 +86,45 @@ Socket connect() throws IOException {
             connectionInfo.getCertificateChain(),
             this.clientConnectorKeyPair.getPrivate());
 
-    String ipAddress = connectionInfo.getIpAddress();
-    if (connectionConfig.getIpType().equals(IpType.PUBLIC)) {
-      ipAddress = connectionInfo.getPublicIpAddress();
+    String address;
+    switch (connectionConfig.getIpType()) {
+      case PUBLIC:
+        address = connectionInfo.getPublicIpAddress();
+        break;
+      case PSC:
+        // DNS names always end with a period (.), so remove it.
+        address = connectionInfo.getPscDnsName().replaceFirst("\\.$", "");
+        break;
+      default:
+        address = connectionInfo.getIpAddress();
+        break;
     }
 
-    if (ipAddress == null || ipAddress.isEmpty()) {
-      throw new RuntimeException("No IP address for the instance");
+    if (address == null || address.isEmpty()) {
+      throw new RuntimeException(
+          String.format(
+              "Instance does not have an address matching type: %s", connectionConfig.getIpType()));
+    }
+
+    // Use the instance's address as a HostName.
+    String serverName = address;
+    // TODO: use the correct address as server name once PSC DNS is populated
```

Comment:
Shall we file an issue for this and link to it here?

Reply:
Added.